go-controller/pkg/ovn/gatewayrouter/policybasedroutes.go

```diff
@@ -0,0 +1,209 @@
+package gatewayrouter
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	utilnet "k8s.io/utils/net"
+
+	"github.com/ovn-org/libovsdb/client"
+
+	libovsdbops "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/libovsdb/ops"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+	ovntypes "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+type PolicyBasedRoutesManager struct {
+	clusterRouterName string
+	netInfo           util.NetInfo
+	nbClient          client.Client
+}
+
+func NewPolicyBasedRoutesManager(nbClient client.Client, clusterRouterName string, netInfo util.NetInfo) *PolicyBasedRoutesManager {
+	return &PolicyBasedRoutesManager{
+		clusterRouterName: clusterRouterName,
+		netInfo:           netInfo,
+		nbClient:          nbClient,
+	}
+}
+
+func (pbr *PolicyBasedRoutesManager) Add(nodeName, mgmtPortIP string, hostIfCIDR *net.IPNet, otherHostAddrs []string) error {
+	var l3Prefix string
+	if utilnet.IsIPv6(hostIfCIDR.IP) {
+		l3Prefix = "ip6"
+	} else {
+		l3Prefix = "ip4"
+	}
+
+	matches := sets.New[string]()
+	for _, hostIP := range append(otherHostAddrs, hostIfCIDR.IP.String()) {
+		// embed nodeName as comment so that it is easier to delete these rules later on.
+		// logical router policy doesn't support external_ids to stash metadata
```

Comment:
id say remove the second comment :P 
`// logical router policy doesn't support external_ids to stash metadata` I know its copy paste artifact.

Reply:
ignoring